app/data/models/application.js

```diff
@@ -1,9 +1,25 @@
-module.exports = (sequelize, DataTypes) => sequelize.define('application', {
-  applicationId: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
-  reference: DataTypes.STRING
-},
-{
-  tableName: 'application',
-  freezeTableName: true,
-  timestamps: false
-})
+module.exports = (sequelize, DataTypes) => {
+  return sequelize.define('application', {
+    id: {
+      type: DataTypes.BIGINT,
+      primaryKey: true,
+      autoIncrement: true
+    },
+    reference: {
+      type: DataTypes.STRING,
+      primaryKey: true,
+      set (val) {
+        this.setDataValue('reference', val.toUpperCase())
+      }
+    },
+    type: { type: DataTypes.STRING, primaryKey: true, allowNull: false },
+    data: DataTypes.STRING,
+    createdAt: { type: DataTypes.DATE, defaultValue: Date.now() },
+    updatedAt: { type: DataTypes.DATE, defaultValue: Date.now() },
```

Comment:
Should `updatedAt` default to null? I would have thought the first time it is created it would only set `createdAt`.

Reply:
yes. Can be .